docs/odata-protocol/odata-protocol.md

```diff
@@ -4722,25 +4743,57 @@ with the specified media stream if it does not already exist, otherwise the
 media stream of the existing media entity is
 [updated](#UpdateaMediaEntityStream).
 
-Singleton entities can be upserted if they are nullable. Services
-supporting this SHOULD advertise it by annotating the singleton with the
-term `Capabilities.UpdateRestrictions` (nested property `Upsertable`
-with value `true`) defined in [OData-VocCap](#ODataVocCap).
-
 A key property whose value is provided in the request URL SHOULD be omitted from the request body.
 If key properties are provided in the request URL and the request body with different values,
 services MUST either fail the request or ignore the value in the request body.
 
-To ensure that an update request is not treated as an insert, the client
-MAY specify an [`If-Match`](#HeaderIfMatch) header in the update
-request. The service MUST NOT treat an update request containing an
-`If-Match` header as an insert.
+If the URL of an update request falls into case 1, 2, or 3 given above and the request
+- contains an [`If-Match: *`](#HeaderIfMatch) header, the service MUST fail the request
+  if an upsert would occur
+- contains an `If-Match` header with an ETag as value, the service fails the request
+  instead of performing an upsert because the [ETag](#UseofETagsforAvoidingUpdateConflicts)
+  does not match
+- contains an [`If-None-Match: *`](#HeaderIfNoneMatch) header, the service MUST fail the request
+  if no upsert would occur.
+
+In other words, the `If-`(`None-`)`Match` header distinguishes between upsert
+and no upsert in these cases.
+
+Alternatively, the URL of an update request ends
+with a non-containment navigation property, optionally followed by a key predicate.
+In this case, the `If-`(`None-`)`Match` header distinguishes between cases where
+the related entity exists and where not, but this is not
+necessarily equivalent to the upsert distinction (see also
+[example 85](#specialization)). If the request
+- contains an `If-Match: *` header, the service MUST fail the request
+  if the navigation property is null
+- contains an `If-Match` header with an ETag as value, the service fails the request
+  if the navigation property is null because the ETag does not match
+- contains an `If-None-Match: *` header, the service MUST fail the request
+  if the navigation property is not null.
+
+Note that if the service does not fail here, it may update the navigation property to point to
```

Comment:
Which leaves the question whether to check [Updatable](https://github.com/oasis-tcs/odata-vocabularies/blob/main/vocabularies/Org.OData.Capabilities.V1.md#UpdateRestrictionsType) or Upsertable before sending such a request.